### PR TITLE
chore: Refactor Large UI components.

### LINKED ISSUE #12600: [CAL-2777] Refactor large UI components
To improve the modularity, testability, and maintainability of our UI, we propose refactoring these large components into smaller, more focused, and "dumb" components. 

### Desired Features:

- **Modular Refactoring**: Break down large UI components into smaller, self-contained, and modular "dumb" components that focus on specific responsibilities.
- **Clear Separation of Concerns**: Ensure that each refactored component has a clear and singular responsibility, promoting the separation of concerns.
- **Reusability**: Design components to be easily reusable across different parts of the application, fostering a more efficient and consistent user interface.
- **Testing Support**: Facilitate unit testing by creating components that are independent of business logic, making it easier to mock and test in isolation.
- **Documentation**: Provide clear documentation for developers, explaining the purpose and usage of each refactored component.

<sub>[CAL-2777](https://linear.app/calcom/issue/CAL-2777/refactor-large-ui-components)</sub>

### PR DESCRIPTION
**Pull request by PIG208**
## What does this PR do?

This progressively breaks down large components into smaller one and aims at making similar future refactoring easier.

- Refactored `BookListItem` and broke it down into a component responsible for all the actions. Also stripped away the standalone components found in the same file.
- Deduplicate code in `customer.ts` that handles stripe payment.

Fixes #12600

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

No behavioral change is expected from this PR, so the existing test cases are expected to pass.

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by PIG208** [comment]
The current sequence of commits is ready for review. More refactoring to be added.

**Review by joeauyeung — COMMENTED** [review]
Looking good so far. Just left one suggestion.

File: apps/web/components/booking/BookingListItemActions.tsx [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+}
+
+function BookingListItemActions({
+  booking,
+  setChargeCardDialogIsOpen,
+  setViewRecordingsDialogIsOpen,
+  setIsOpenRescheduleDialog,
+  setIsOpenLocationDialog,
+  showViewRecordingsButton,
+  showCheckRecordingButton,
```
[the remark this answers is not in the document]
Comment (PIG208) [reply apps/web/components/booking/BookingListItemActions.tsx:55]: Thanks for the review. Do you mean that payment actions and view recordings should have their own separate component?

File: apps/web/components/booking/RecurringBookingsTooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        .includes(recurringDate.toString())
+    );
+  }).length;
+
+  return (
+    (booking.recurringInfo &&
+      booking.eventType?.recurringEvent?.freq &&
+      (booking.listingStatus === "recurring" ||
+        booking.listingStatus === "unconfirmed" ||
+        booking.listingStatus === "cancelled") && (
```
Comment (sean-brydon) [inline apps/web/components/booking/RecurringBookingsTooltip.tsx:44]: Can we early return if this conditional is not met? 

Havent tested but something like 

``` 
  const isRecurringBooking = booking.recurringInfo && booking.eventType?.recurringEvent;
  const isRecurringListingStatus = ["recurring", "unconfirmed", "cancelled"].includes(booking.listingStatus);
  if (
    !isRecurringBooking ||
    !isRecurringListingStatus ||
    !booking.eventType.recurringEvent ||
    !booking.recurringInfo?.count
  ) {
    return null;
  }
``` 
should work

**Review by sean-brydon — CHANGES_REQUESTED** [review]
Overall LGTM - lets just tidy up that tooltip component when we are here. 

I will create a follow up issue to extract the Dialogs out of each list item.

**Pull request closed without merging**